### PR TITLE
address concerns that depwarn for transpose is unclear

### PR DESCRIPTION
**Pull request by cjprybol**
I believe this synthesizes all of the concerns expressed in https://github.com/JuliaLang/julia/issues/18320

### CODE REVIEW

**Review by tkelman — APPROVED** [review]
Looks okay to me, thanks.

Anyone else have comments here?

edit: could use a description of what's being fixed in the commit message, whoever does "merge and squash" can add a few words to the message at merge time if needed

**Comment by andreasnoack** [comment]
LGTM

File: base/deprecated.jl [hunk 1]
```diff
@@ -765,9 +765,10 @@ end
 
 # Deprecate no-op transpose fallback. Please see #13171 and #17075.
 function transpose(x)
-    depwarn(string("the no-op `transpose` fallback is deprecated, and no more specific ",
-        "`transpose` method for $(typeof(x)) exists. Consider `permutedims(x, [2, 1])` ",
-        "or writing a specific `transpose(x::$(typeof(x)))` method if appropriate."),
+    depwarn(string("the no-op `transpose` for non-numeric arrays is deprecated, ",
+        "and no specific `transpose` method for $(typeof(x)) exists. Use ",
+        "permutedims(x, (2, 1)) for matrices and reshape(x, 1, length(x)) for vectors, ",
```
Comment (ararslan) [inline base/deprecated.jl]: Should preserve the backticks around `permutedims(...)` and `reshape(...)`.

**Review by ararslan — CHANGES_REQUESTED** [review]

File: base/deprecated.jl (hunk 1 above)
Comment (cjprybol) [inline base/deprecated.jl]: Good eye, thanks!

**Pull request merged**